saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/classifier/SL_model/Saul_SL_Inference.scala

```diff
@@ -95,11 +96,13 @@ class Saul_SL_Inference[HEAD <: AnyRef](factors: List[ConstrainedClassifier[_, H
             //    println("size mismatch!", cf.onClassifier.classifier.asInstanceOf[SparseNetworkLBP].getNetwork.get(i).asInstanceOf[LinearThresholdUnit].getWeightVector.size(), ",", ltuSize.size)
             //              }
             val exampleFeatureIndexes = ofDim[Int](myLTUJointlyTrainedWeight.length)
-            cf.onClassifier.classifier.asInstanceOf[SparseNetworkLBP].getLTU(i).getWeightVector.clear()
+            val weightVector = cf.onClassifier.classifier.asInstanceOf[SparseNetworkLearner].getLTU(i).asInstanceOf[LinearThresholdUnit].getWeightVector
+            weightVector.clear()
+
             // val  exampleFeatureIndexes = cf.onClassifier.classifier.asInstanceOf[SparseNetworkLBP].getLexicon.getMap.values.toArray.map(_.asInstanceOf[Int])//.toArray//.toArray().asInstanceOf[Array[Int]]
-            for (featureIndex <- 0 until myLTUJointlyTrainedWeight.length) {
+            for (featureIndex <- myLTUJointlyTrainedWeight.indices) {
               exampleFeatureIndexes(featureIndex) = featureIndex
-              cf.onClassifier.classifier.asInstanceOf[SparseNetworkLBP].getLTU(i).getWeightVector.setWeight(featureIndex, myLTUJointlyTrainedWeight(featureIndex))
+              weightVector.setWeight(featureIndex, myLTUJointlyTrainedWeight(featureIndex))
               //cf.onClassifier.classifier.asInstanceOf[SparseNetworkLBP].getLexicon.;
```

Comment:
same here, I did it directly on cf since setWeight seemed not working. I need to trace these unless you are very sure about it.